### PR TITLE
Adds some useful Tagliatelle macros

### PR DESCRIPTION
**Pull request by D3nnisH**
for methods in `sirius.kernel.commons.Strings`

### CODE REVIEW

File: src/main/java/sirius/tagliatelle/macros/JoinMacro.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import sirius.kernel.di.std.Register;
+import sirius.tagliatelle.Tagliatelle;
+import sirius.tagliatelle.expression.Expression;
+import sirius.tagliatelle.rendering.LocalRenderContext;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+
+/**
+ * Represents <tt>join(Iterable)</tt> which is a call to {@link Strings#urlEncode(String)}.
```
Comment (sabieber) [inline src/main/java/sirius/tagliatelle/macros/JoinMacro.java]: urlEncode -> join

Comment (sabieber) [reply src/main/java/sirius/tagliatelle/macros/JoinMacro.java]: and the second parameter is not described

File: src/main/java/sirius/tagliatelle/macros/ShortenMacro.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import sirius.tagliatelle.rendering.LocalRenderContext;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+
+/**
+ * Represents <tt>shorten(String, int)</tt> which is a call to {@link Strings#shorten(String, int)}.
+ */
+@Register
+public class ShortenMacro implements Macro {
```
Comment (sabieber) [inline src/main/java/sirius/tagliatelle/macros/ShortenMacro.java]: Functionality is already possible with the limit Macro

File: pom.xml [hunk 3]
```diff
@@ -9,7 +9,7 @@
         <version>3.8.3</version>
     </parent>
     <artifactId>sirius-web</artifactId>
-    <version>11.10.2</version>
+    <version>11.10.3</version>
```
Comment (sabieber) [inline pom.xml]: Please upgrade to 11.11

File: src/main/java/sirius/tagliatelle/macros/ShortenMacro.java (hunk 2 above)
Comment (D3nnisH) [reply src/main/java/sirius/tagliatelle/macros/ShortenMacro.java]: but LimitMacro doesn't add ellipsis (3rd parameter is implicit false)

File: src/main/java/sirius/tagliatelle/macros/JoinMacro.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    @Override
+    public Object eval(LocalRenderContext ctx, Expression[] args) {
+        return Strings.join((Iterable<?>) args[0].eval(ctx), (String) args[1].eval(ctx));
+    }
+
+    @Override
+    public boolean isConstant(Expression[] args) {
+        return true;
```
Comment (andyHa) [inline src/main/java/sirius/tagliatelle/macros/JoinMacro.java]: ask the parameters maybe?

File: src/main/java/sirius/tagliatelle/macros/ToListMacro.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        if (args.length == 1 && args[0].getType().isArray()) {
+            return Arrays.stream((Object[]) args[0].eval(ctx)).collect(Collectors.toList());
+        } else {
+            return Arrays.stream(args).map(element -> element.eval(ctx)).collect(Collectors.toList());
+        }
+    }
+
+    @Override
+    public boolean isConstant(Expression[] args) {
+        return true;
```
Comment (andyHa) [inline src/main/java/sirius/tagliatelle/macros/ToListMacro.java]: ask params

**Pull request merged**